File: backend/app/schemas/story.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class RunStoryRequest(BaseModel):
    query: str
    triggered_by: str = "manual"


class RunStoryResponse(BaseModel):
    run_id: str
    story_id: str
    status: str = "discovering"


class ApproveRequest(BaseModel):
    notes: str = ""


class RejectRequest(BaseModel):
    reason: str


class EditRequest(BaseModel):
    field: str
    value: str


class PublishTarget(BaseModel):
    platform: str  # "facebook" | "instagram"
    page_id: str
    scheduled_at: Optional[datetime] = None


class PublishRequest(BaseModel):
    targets: list[PublishTarget]


class StoryRead(BaseModel):
    id: str
    run_id: str
    query: str
    status: str
    confidence_tier: Optional[str] = None
    overall_confidence: Optional[float] = None
    fake_news_flags: Optional[list] = None
    headline: Optional[str] = None
    sub_headline: Optional[str] = None
    summary: Optional[str] = None
    bullet_points: Optional[list] = None
    hashtags: Optional[list] = None
    caption_facebook: Optional[str] = None
    caption_instagram: Optional[str] = None
    selected_image_url: Optional[str] = None
    candidate_images: Optional[list] = None
    social_card_paths: Optional[dict] = None
    seo_data: Optional[dict] = None
    language: str = "en"
    approval_status: str = "pending"
    human_edits: Optional[list] = None
    approved_at: Optional[datetime] = None
    rejection_reason: Optional[str] = None
    triggered_by: str = "manual"
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/story.ts
export interface Story {
  id: string;
  run_id: string;
  query: string;
  status: StoryStatus;
  confidence_tier?: "low" | "medium" | "high";
  overall_confidence?: number;
  fake_news_flags?: string[];
  headline?: string;
  sub_headline?: string;
  summary?: string;
  bullet_points?: string[];
  hashtags?: string[];
  caption_facebook?: string;
  caption_instagram?: string;
  selected_image_url?: string;
  candidate_images?: MediaAsset[];
  social_card_paths?: Record<"facebook" | "instagram", string>;
  seo_data?: SEOData;
  language: string;
  approval_status: ApprovalStatus;
  human_edits?: HumanEdit[];
  approved_at?: string;
  rejection_reason?: string;
  triggered_by: string;
  created_at: string;
  updated_at: string;
}

export type StoryStatus =
  | "discovering"
  | "verifying"
  | "summarizing"
  | "media"
  | "pending_approval"
  | "approved"
  | "rejected"
  | "publishing"
  | "published"
  | "failed";

export type ApprovalStatus = "pending" | "approved" | "rejected" | "editing";

export interface MediaAsset {
  url: string;
  local_path?: string;
  relevance_score: number;
  alt_text: string;
  source_attribution: string;
  is_ai_generated: boolean;
  width: number;
  height: number;
}

export interface SEOData {
  score: number;
  focus_keyword: string;
  secondary_keywords: string[];
  meta_description: string;
  slug: string;
  readability: string;
  tips: string[];
}

export interface HumanEdit {
  field: string;
  original: string;
  edited: string;
  editor_user_id: string;
  edited_at: string;
}
